x/crosschain/keeper/grpc_query_cctx_rate_limit_test.go

```diff
@@ -123,60 +124,486 @@ func setupForeignCoins(
+
+		// convert cctx value
+		value := keeper.ConvertCctxValue(ethChainID, cctx, gasCoinRates, erc20CoinRates, foreignCoinMap)
+		require.Equal(t, sdk.NewDec(0), value)
+	})
+	t.Run("should return 0 on nil rate or rate <= 0", func(t *testing.T) {
+		cctx := sample.CrossChainTx(t, fmt.Sprintf("%d-%d", ethChainID, 1))
+		cctx.InboundTxParams.CoinType = coin.CoinType_Gas
+		cctx.GetCurrentOutTxParam().Amount = sdk.NewUint(100)
+
+		// use nil gasCoinRates map to convert cctx value
+		value := keeper.ConvertCctxValue(ethChainID, cctx, nil, erc20CoinRates, foreignCoinMap)
+		require.Equal(t, sdk.NewDec(0), value)
+
+		// set rate to 0
+		zeroCoinRates, _ := k.GetRateLimiterRates(ctx)
+		zeroCoinRates[ethChainID] = sdk.NewDec(0)
+
+		// convert cctx value
+		value = keeper.ConvertCctxValue(ethChainID, cctx, zeroCoinRates, erc20CoinRates, foreignCoinMap)
+		require.Equal(t, sdk.NewDec(0), value)
+
+		// set rate to -1
+		negativeCoinRates, _ := k.GetRateLimiterRates(ctx)
+		negativeCoinRates[ethChainID] = sdk.NewDec(-1)
+
+		// convert cctx value
+		value = keeper.ConvertCctxValue(ethChainID, cctx, negativeCoinRates, erc20CoinRates, foreignCoinMap)
+		require.Equal(t, sdk.NewDec(0), value)
+	})
+	t.Run("should return 0 if no coin found for chainID", func(t *testing.T) {
+		cctx := sample.CrossChainTx(t, fmt.Sprintf("%d-%d", ethChainID, 1))
+		cctx.InboundTxParams.CoinType = coin.CoinType_Gas
+		cctx.GetCurrentOutTxParam().Amount = sdk.NewUint(100)
+
+		// use empty foreignCoinMap to convert cctx value
+		value := keeper.ConvertCctxValue(ethChainID, cctx, gasCoinRates, erc20CoinRates, nil)
+		require.Equal(t, sdk.NewDec(0), value)
+	})
+	t.Run("should return 0 if no coin found for asset", func(t *testing.T) {
+		cctx := sample.CrossChainTx(t, fmt.Sprintf("%d-%d", ethChainID, 1))
+		cctx.InboundTxParams.CoinType = coin.CoinType_ERC20
+		cctx.InboundTxParams.Asset = assetUSDT
+		cctx.GetCurrentOutTxParam().Amount = sdk.NewUint(100)
+
+		// delete assetUSDT from foreignCoinMap for ethChainID
+		tempCoinMap := zk.FungibleKeeper.GetAllForeignCoinMap(ctx)
+		delete(tempCoinMap[ethChainID], strings.ToLower(assetUSDT))
+
+		// convert cctx value
+		value := keeper.ConvertCctxValue(ethChainID, cctx, gasCoinRates, erc20CoinRates, tempCoinMap)
+		require.Equal(t, sdk.NewDec(0), value)
+	})
+}
+
+func TestKeeper_ListPendingCctxWithinRateLimit(t *testing.T) {
+	// create sample TSS
+	tss := sample.Tss()
+
+	// create sample zrc20 addresses for ETH, BTC, USDT
+	zrc20ETH := sample.EthAddress().Hex()
+	zrc20BTC := sample.EthAddress().Hex()
+	zrc20USDT := sample.EthAddress().Hex()
+
+	// create Eth chain mined and pending cctxs for rate limiter test
+	ethMindedCctxs := createCctxsWithCoinTypeAndHeightRange(t, 1, 999, ethChainID, coin.CoinType_Gas, "", uint64(1e15), types.CctxStatus_OutboundMined)
+	ethPendingCctxs := createCctxsWithCoinTypeAndHeightRange(t, 1000, 1199, ethChainID, coin.CoinType_Gas, "", uint64(1e15), types.CctxStatus_PendingOutbound)
```

Comment:
Any reason choosing 1000 and 1199?
We might want to precise in comment what these values represent

Reply:
Let me add a few descriptions

Reply:
https://github.com/zeta-chain/node/pull/2060/commits/578afccb54c86839c9c505dab05893132164b7ee